src/Object/Photo.php

```diff
@@ -0,0 +1,1206 @@
+<?php
+/**
+ * @file src/Object/Photo.php
+ * @brief This file contains the Photo class for image processing
+ */
+namespace Friendica\Object;
+
+use Friendica\App;
+use Friendica\Core\Cache;
+use Friendica\Core\Config;
+use Friendica\Core\System;
+use Friendica\Database\DBM;
+use dba;
+
+require_once "include/photos.php";
+
+/**
+ * Class to handle Photos
+ */
+class Photo
+{
+	private $image;
+
+	/*
+	 * Put back gd stuff, not everybody have Imagick
+	 */
+	private $imagick;
+	private $width;
+	private $height;
+	private $valid;
+	private $type;
+	private $types;
+
+	/**
+	 * @brief supported mimetypes and corresponding file extensions
+	 */
+	public static function supportedTypes()
+	{
+		if (class_exists('Imagick')) {
+			// Imagick::queryFormats won't help us a lot there...
+			// At least, not yet, other parts of friendica uses this array
+			$t = array(
+				'image/jpeg' => 'jpg',
+				'image/png' => 'png',
+				'image/gif' => 'gif'
+			);
+		} else {
+			$t = array();
+			$t['image/jpeg'] ='jpg';
+			if (imagetypes() & IMG_PNG) {
+				$t['image/png'] = 'png';
+			}
+		}
+
+		return $t;
+	}
+
+	/**
+	 * @brief Constructor
+	 * @param object  $data data
+	 * @param boolean $type optional, default null
+	 * @return object
+	 */
+	public function __construct($data, $type = null)
+	{
+		$this->imagick = class_exists('Imagick');
+		$this->types = static::supportedTypes();
+		if (!array_key_exists($type, $this->types)) {
+			$type='image/jpeg';
+		}
+		$this->type = $type;
+
+		if ($this->isImagick() && $this->loadData($data)) {
+			return true;
+		} else {
+			// Failed to load with Imagick, fallback
+			$this->imagick = false;
+		}
+		return $this->loadData($data);
+	}
+
+	/**
+	 * @brief Destructor
+	 * @return void
+	 */
+	public function __destruct()
+	{
+		if ($this->image) {
+			if ($this->isImagick()) {
+				$this->image->\Imagick::clear();
+				$this->image->\Imagick::destroy();
+				return;
+			}
+			if (is_resource($this->image)) {
+				imagedestroy($this->image);
+			}
+		}
+	}
+
+	/**
+	 * @return boolean
+	 */
+	public function isImagick()
+	{
+		return $this->imagick;
+	}
+
+	/**
+	 * @brief Maps Mime types to Imagick formats
+	 * @return arr With with image formats (mime type as key)
+	 */
+	public function getFormatsMap()
+	{
+		$m = array(
+			'image/jpeg' => 'JPG',
+			'image/png' => 'PNG',
+			'image/gif' => 'GIF'
+		);
+		return $m;
+	}
+
+	/**
+	 * @param object $data data
+	 * @return boolean
+	 */
+	private function loadData($data)
+	{
+		if ($this->isImagick()) {
+			$this->image = new \Imagick();
```

Comment:
Why not add a couple of `use` statements? Wouldn't it remove the need for the horrible syntax like `$this->image->\Imagick::readImageBlob($data);`?

Reply:
Can you give me an example of how? I am having trouble getting this to work...

Reply:
Just add 
```php
use Imagick;
use ImagickPixel;
```
After the other `use` statements, and you don't have to add the backslashes in front of the `Imagick` class names. Not sure about the individual method calls.